tests/compiler/std/array.ts

```diff
@@ -978,5 +978,30 @@ class Ref { constructor() {} }
   assert(subarrU32.toString() == "1");
 }
 
+// Array#flat //////////////////////////////////////////////////////////////////////////////////
+{
+  var flatTarget: i32[][] = [[0], [1, 2, 3], [4, 5, 6], [7, 8, 9]];
+  var result = flatTarget.flat();
+  for (let i = 0; i < 10; i++) {
+    assert(result[i] == i, (i).toString());
+  }
+  assert(result.length == 10);
+
+  var flatStringTarget: string[][] = [["one"], ["two", "three"], ["four", "five", "six"], null, ["seven"]];
+  var stringResult = flatStringTarget.flat();
+  assert(stringResult[0] == "one");
+  assert(stringResult[1] == "two");
+  assert(stringResult[2] == "three");
+  assert(stringResult[3] == "four");
+  assert(stringResult[4] == "five");
+  assert(stringResult[5] == "six");
+  assert(stringResult[6] == null);
+  assert(stringResult[7] == "seven");
+  assert(stringResult.length == 8);
+
+  let testArray: i32[][] = [[], [], [], [], [], [], [], []];
```

Comment:
I guess just `[[], []]` enough =)

Reply:
Is this really necessary to fix? I added the string cases, and I don't see a good reason to change this.

Reply:
No really necessary, of course. So up to you